endpoints/openrtb2/amp_auction_test.go

```diff
@@ -122,357 +122,291 @@ func TestAMPPageInfo(t *testing.T) {
 	assert.Equal(t, "test.somepage.co.uk", exchange.lastRequest.Site.Domain)
 }
 
-func TestConsentThroughEndpoint(t *testing.T) {
-	// gdpr consent string that will come inside our http.Request query
-	const consentString = "BOa71ZYOa71ZYAbABBENA8-AAAAbN7_______9______9uz_Gv_r_f__33e8_39v_h_7_-___m_-3zV4-_lvR11yPA1OrfIrwFhiAw"
-	const DigiTurstID = "digitrustId"
-
-	// Generate a marshaled openrtb.BidRequest that DOESN'T come with a gdpr consent string
-	fullMarshaledBidRequest, err := getTestBidRequest(false, false, "", DigiTurstID)
-	if err != nil {
-		t.Fatalf("Failed to marshal the complete openrtb.BidRequest object %v", err)
-	}
-
-	stored := map[string]json.RawMessage{
-		"1": json.RawMessage(fullMarshaledBidRequest),
-	}
-
-	theMetrics := pbsmetrics.NewMetrics(metrics.NewRegistry(), openrtb_ext.BidderList(), config.DisabledMetrics{})
-	exchange := &mockAmpExchange{}
-
-	endpoint, _ := NewAmpEndpoint(
-		exchange,
-		newParamsValidator(t),
-		&mockAmpStoredReqFetcher{stored},
-		empty_fetcher.EmptyFetcher{},
-		&config.Configuration{MaxRequestSize: maxSize},
-		theMetrics,
-		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
-		map[string]string{},
-		[]byte{},
-		openrtb_ext.BidderMap,
-	)
-	request := httptest.NewRequest("GET", fmt.Sprintf("/openrtb2/auction/amp?tag_id=1&gdpr_consent=%s", consentString), nil)
-	recorder := httptest.NewRecorder()
-	endpoint(recorder, request, nil)
-
-	// Assert our bidRequest was valid
-	if !assert.NotNil(t, exchange.lastRequest, "Endpoint responded with %d: %s", recorder.Code, recorder.Body.String()) {
-		return
-	}
-	// Assert our bidRequest had a valid "User" field
-	if !assert.NotNil(t, exchange.lastRequest.User, "Resulting bid request should have a valid User field after passing consent string through endpoint") {
-		return
-	}
-	// Assert our bidRequest had a valid "User.Ext" field
-	if !assert.NotNil(t, exchange.lastRequest.User.Ext, "Resulting bid request should have a valid Ext field after passing consent string through endpoint") {
-		return
-	}
-
-	// Assert string `consent` is found in the User.Ext at all
-	assert.NotContainsf(t, fullMarshaledBidRequest, "consent:"+consentString, "Expected bid request to contain consent string %s \n", consentString)
-
-	// Assert the last request has a valid User object with a consent string equal to that on the URL query
-	var ue openrtb_ext.ExtUser
-	err = json.Unmarshal(exchange.lastRequest.User.Ext, &ue)
-	assert.NoError(t, err, "Error unmarshalling last processed request")
-
-	// Assert consent string found in `http.Request` was passed correctly to the `User.Ext` object
```

Comment:
I see the following assertions were removed. Do you think they are worth checking?
```
    // Assert consent string found in `http.Request` was passed correctly to the `User.Ext` object
    assert.Contains(t, string(request.URL.RawQuery), consentString, "http.Request should come with a consent string in its query")
    assert.Equal(t, consentString, ue.Consent, "Consent string unsuccessfully passed to bid request through AMP endpoint")

    // Assert other user properties found originally in our bid request such as `DigiTrust` were not overwritten
    assert.Equal(t, DigiTurstID, ue.DigiTrust.ID, "Passing GDPR consent through endpoint should not override http.Request ExtUser fields other than consent")
amp_auction_test.go
```

Reply:
Those checks are (mostly) still present, just in different forms. The new tests assert the entire User.Ext object instead of breaking it down into individual assertions. That just leaves...

> assert.Contains(t, string(request.URL.RawQuery), consentString, "http.Request should come with a consent string in its query")

Which looks to me that it's asserting the test setup and is therefore not useful.